### ISSUE TITLE
I have with listen on some why in the UDP protocol with codenvy?

### ISSUE THREAD

**Issue by PerduGames** [codenvy/codenvy#2529]
I have with listen on some why in the UDP protocol with codenvy?

**Comment by eivantsov** [comment]
@PerduGames please edit your issue according to a template (you should have seen one when you created an issue), otherwise this issue will be closed.

**Comment by PerduGames** [comment]
@eivantsov edited.

**Comment by eivantsov** [comment]
@PerduGames I still don;t understand what the problem is. Do you want to get preview URL for your server?

**Comment by PerduGames** [comment]
@eivantsov I want to know if I can listen on the UDP port using codenvy.

**Comment by eivantsov** [comment]
@PerduGames when adding servers in User Dashboard, you can specify protocol

**Comment by PerduGames** [comment]
Then it is possible to find the configuration here, accepted when I typed udp and ecolhi a port to create the server. I hope it works. Thanks for the help.

**Comment by PerduGames** [comment]
I'm trying to listen on port 1507, but I can not, the steps I'm following:

http://i.imgur.com/SX0MkCO.png

```
// CREATE SERVER UDP
const PORT_UDP = 1507
const dgram = require('dgram');
const server = dgram.createSocket('udp4');

server.on('error', (err) => {
    console.log(`server error:\n${err.stack}`);
    server.close();
});

server.on('message', (msg, rinfo) => {
    console.log(`message: ${msg} from ${rinfo.address}:${rinfo.port}`);
  
});

server.on('listening', () => {
    const address = server.address();
    console.log(`server listening ${address.address}:${address.port}`);
  
});
server.bind({
    port: PORT_UDP,
    exclusive: true
    }, () => {  
        console.log('BIND');
});
```
on the client side I try to send messages to the server and it does not get to the server, I'm getting ip from the server with "curl ifconfig.me" on the dev-machine terminal, and sending to port 1507.

**Comment by eivantsov** [comment]
When the application is running, if you go back to servers tab, you will find the actual published port.

**Comment by PerduGames** [comment]
on "RUNTIME"? Nothing appears to me.

http://i.imgur.com/SX0MkCO.png

**Comment by eivantsov** [comment]
Your workspace should be running, then you will see this:

![image](https://user-images.githubusercontent.com/5337267/32978777-a86f2548-cc51-11e7-8d9a-28e2dc426a7d.png)

**Comment by PerduGames** [comment]
But I'm not seeing, when it's running or not, it's still empty on **RUNTIME**.

**Comment by eivantsov** [comment]
1. When workspace is stopped - add a server
2. Run your workspace
3. Share the same screenshot that I have shared so that it's obvious that your workspace is running

**Comment by PerduGames** [comment]
On here:
![codenvy-not-server](https://user-images.githubusercontent.com/29399008/33020175-62449bc2-cde4-11e7-91ee-b1a7e14f706b.png)

**Comment by PerduGames** [comment]
I changed from 15007 to 1507 and it worked now. So the port that I should put on the client is **44378** and the address to connect to **node11.codenvy.io**?

![codenvy2](https://user-images.githubusercontent.com/29399008/33020355-2b77ab92-cde5-11e7-9eb6-f49612b97793.png)

**Comment by eivantsov** [comment]
@PerduGames yes, but this url changes when you restart a workspace. You can get the url in a preview in the IDE http://www.eclipse.org/che/docs/ide/previews/index.html
